lib/temporal/workflow/context.rb

```diff
@@ -279,7 +280,7 @@ def wait_until(&unblock_condition)
         fiber = Fiber.current
         blocked = true
```

Comment:
Do we still need this flag if we are now unregistering event handlers upon resuming?

Reply:
I think you're right. At first, I wasn't sure if there were callback within callback cases where this wouldn't work. But it should behave deterministically. As soon as a handler block is invoked for the first time, its fiber will be resumed, and unregistration will be completed before any other handler blocks are invoked.